packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/GeoSpatialTests.kt

```diff
@@ -0,0 +1,411 @@
+
+        val onlyOuterRing = GeoPolygon(
+            outerRing = listOf(
+                GeoPoint(-5.0, -5.0),
+                GeoPoint(5.0, -5.0),
+                GeoPoint(5.0, 5.0),
+                GeoPoint(-5.0, 5.0),
+                GeoPoint(-5.0, -5.0)
+            )
+        )
+        assertEquals(2, realm.query<Restaurant>("location GEOWITHIN $onlyOuterRing").count().find())
+
+        val polygonWithHole = GeoPolygon(
+            outerRing = listOf(
+                GeoPoint(-5.0, -5.0),
+                GeoPoint(5.0, -5.0),
+                GeoPoint(5.0, 5.0),
+                GeoPoint(-5.0, 5.0),
+                GeoPoint(-5.0, -5.0)
+            ),
+            hole = listOf(
+                GeoPoint(-4.0, -4.0),
+                GeoPoint(4.0, -4.0),
+                GeoPoint(4.0, 4.0),
+                GeoPoint(-4.0, 4.0),
+                GeoPoint(-4.0, -4.0)
+            )
+        )
+        assertEquals(1, realm.query<Restaurant>("location GEOWITHIN $polygonWithHole").count().find())
+    }
+
+    @Test
+    fun geoPolygon_within_throwsOnNonClosedPolygon() {
+        val polygon1 = GeoPolygon(
+            outerRing = listOf(
+                GeoPoint(-5.0, -5.0),
+                GeoPoint(5.0, -5.0),
+                GeoPoint(5.0, 5.0)
+            )
+        )
+        assertFailsWithMessage<IllegalArgumentException>("Invalid region in GEOWITHIN query") {
+            realm.query<Restaurant>("location GEOWITHIN $polygon1").count()
+        }
+    }
+
+    @Test
+    fun distance_radians() {
+        val validDists = listOf(0.0, 0.1, Double.MIN_VALUE, Double.MAX_VALUE, Double.POSITIVE_INFINITY)
+        validDists.forEach { d: Double ->
+            val dist1 = Distance.fromRadians(d)
+            val dist2 = d.radians
+            assertEquals(d, dist1.inRadians)
+            assertEquals(d, dist2.inRadians)
+        }
+    }
+
+    @Test
+    fun distance_kilometer() {
+        val validDists = mapOf(
+            0.0 to 0.0,
+            0.1 to 0.1,
+            Double.MIN_VALUE to 0.0,
+            Double.MAX_VALUE to Double.MAX_VALUE,
+            Double.POSITIVE_INFINITY to Double.POSITIVE_INFINITY
+        )
+        validDists.forEach { (input, output) ->
+            val dist1 = Distance.fromKilometers(input)
+            val dist2 = input.km
+            assertEquals(output, dist1.inKilometers, "$input failed.")
+            assertEquals(output, dist2.inKilometers, "$input failed.")
+        }
+    }
+
+    @Test
+    fun distance_miles() {
+        val validDists = mapOf(
+            0.0 to 0.0,
+            0.1 to 0.10000000000000002,
+            Double.MIN_VALUE to 0.0,
+            Double.MAX_VALUE to Double.POSITIVE_INFINITY,
+            Double.POSITIVE_INFINITY to Double.POSITIVE_INFINITY
+        )
+        validDists.forEach { (input, output) ->
+            val dist1 = Distance.fromMiles(input)
+            val dist2 = input.miles
+            assertEquals(output, dist1.inMiles, "$input failed.")
```

Comment:
Same. Not ok to test round tripped values. Need to test some visual recognizable values.